lib/handlebars/compiler/javascript-compiler.js

```diff
@@ -638,10 +674,27 @@ JavaScriptCompiler.prototype = {
         '(', lookup,
         (helper.paramsInit ? ['),(', helper.paramsInit] : []), '),',
         '(typeof helper === ', this.aliasable('"function"'), ' ? ',
-        this.source.functionCall('helper', 'call', helper.callParams), ' : helper))'
+        this.helperFunctionCall('helper', helper, null), ' : helper))'
     ]);
   },
 
+  helperFunctionCall: function(helperName, helperOptions, splatMap) {
+    if (splatMap) {
+      let splatMapObj = {};
```

Comment:
Any thoughts on performance/code size of this vs. array with missing values notation? I.e. `[,,1]` vs. `{"2":1}`.

Reply:
I don't know, don't have the perf wizard knowledge myself.